### PR TITLE
Added flake8 travis tests and badge

### PR DESCRIPTION
**Pull request by patcon**
No corrections so far, in case there are some we don't care about. So far, just ignoring the max line length.

If I find a good tool to auto-correct the issues, i might send a PR before I hear back which rules we're interested in enforcing :)

### CODE REVIEW

**Comment by patcon** [comment]
Used `autopep8` to correct a bunch of things, then manually addressed the others. Ignoring line length checks, and also ignoring a check related to lambda's, which seemed a bit invasive.

The tests should pass as-is, and this is ready for review when you get time

**Comment by patcon** [comment]
all your rebase are belong to us

File: councilmatic_core/views.py [hunk 1]
```diff
@@ -99,8 +99,6 @@ class IndexView(TemplateView):
     event_model = Event
 
     def get_context_data(self, **kwargs):
-        context = super(IndexView, self).get_context_data(**kwargs)
```
Comment (cathydeng) [inline councilmatic_core/views.py]: we need this to get city-specific overrides of the homepage to work - this allows each city to have a custom context. can you add this back in? everything else looks like it's merely a style change, & looks fine

Comment (cathydeng) [inline councilmatic_core/views.py]: is this something flake8 flagged as a style issue? if so, what'd it say?

Comment (patcon) [inline councilmatic_core/views.py]: Hm... yeah, it seemed odd remove, so i flagged it in it's own commit: F841
http://flake8.pycqa.org/en/latest/warnings.html?highlight=F841

> local variable name is assigned to but never used

All the other context variables were returned, and this one wasn't. Should it have been?

Seems the context object isn't actually used or changed in the `IndexView`, though perhaps I'm missing something. Perhaps the returned object in that view should be merged with that context var?

**Comment by patcon** [comment]
OK, not sure if it was causing problems, but seemed that we weren't actually editting the context object, but were instead returning a fresh dict. Seems there was no consequence, but for consistency, we're new merging the generated dict into it, and returning the whole shbang :)

(Only modified the final bugfix commit.)

**Comment by cathydeng** [comment]
ok, testing this out & everything looks fine. wana add a sentence to the readme, maybe another bonus point in the pull requests section, on using flake8 to check styling when developing?

**Comment by patcon** [comment]
Thanks! Done and done :)

**Comment by cathydeng** [comment]
cool, looks good. I don't wana be too strict about styling & require all new folks to adhere to pep8 guidelines in order to contribute, but I'll just make those changes to the readme

**Comment by patcon** [comment]
:+1: I'm sure they're voluntarily become style-junkies when pesky little X's start showing up in their PRs -- no hassling required on your part :)

**Pull request merged**